### PR TITLE
Remove all event listeners from gd

### PR DESCRIPTION
**Pull request by mdtusz**
Piece of :cake:

### CODE REVIEW

File: src/plot_api/plot_api.js [hunk 1]
```diff
@@ -941,6 +941,7 @@ Plotly.redraw = function(gd) {
 Plotly.newPlot = function (gd, data, layout, config) {
     gd = getGraphDiv(gd);
     plots.purge(gd);
+    gd.removeAllListeners();
```
Comment (etpinard) [inline src/plot_api/plot_api.js]: I'd would put this in `plots.purge` 

and maybe (cc @alexcjohnson ) we could expose `purge` on `Plotly`? It sounds to me like something users might want to use.

Comment (mdtusz) [inline src/plot_api/plot_api.js]: Good call on moving `removeAllListeners`. Not sure if we should expose the purge method though - I can't think of too many use cases where that's what a user would want to clear the node and reuse it - if they want to "purge", they'll just delete the node. Better to keep the user surface area of plotly.js as small as possible. 

@alexcjohnson thoughts?

Comment (etpinard) [inline src/plot_api/plot_api.js]: @mdtusz 

> if they want to "purge", they'll just delete the node.

You got me convinced :+1:

Comment (mdtusz) [inline src/plot_api/plot_api.js]: purge it all

**Comment by alexcjohnson** [comment]
I think there are still some client installations that use `plots.purge` because they were written before we had `Plotly.newPlot`... but yeah, I'm good with hiding `plots.purge` anyway and if those old clients ever upgrade plotly.js they can deal, it's probably only 2 or 3 of them. - if you *don't* follow `purge` up with another `Plotly.plot` all hell breaks loose.

**Comment by emackey** [comment]
I'd like to use something similar to `purge` or `destroy` to get rid of a chart that is being dismissed from the UI of a single-page web app. The chart may have been in a popup or overlay, and that's going away now, so I want to kill off the render loop, the event handlers, and indeed get rid of any complex DOM structure that is no longer needed.

**Comment by etpinard** [comment]
:dancer:

**Pull request merged**